src/components/DiscoverModal/DiscoverModal.vue

```diff
@@ -24,12 +24,58 @@ export default {
   data() {
     return {
       validate: false,
-      favorite: false
+      favorite: false,
+      localUser: []
     }
   },
   props: {
     content: { type: Object, required: true },
     dialog: { type: Boolean, default: false }
+  },
+  created() {
+    this.localUser = this.getLocalUser()
+  },
+  updated() {
+    if (this.getLocalUser() !== null) {
+      let localUser = this.getLocalUser()
+      let favList = localUser.favorite_articles
+      const whiteList = favList.map(fav => fav.id)
+      if (whiteList.includes(this.content.id)) {
+        this.favorite = true
+      } else this.favorite = false
+    }
+  },
+  methods: {
+    getLocalUser() {
+      return JSON.parse(localStorage.getItem("userLog"))
+    },
+    setLocalUser(newLocaluser) {
+      localStorage.setItem("userLog", JSON.stringify(newLocaluser))
+    },
+    async setBackUser({ favorite_articles }) {
+      const idUser = this.$store.state.accountData.id
+      // eslint-disable-next-line
+      const sendUser = await this.$http
+        .put(`/user/${idUser}`, { favoriteArticles: favorite_articles })
+        // eslint-disable-next-line
+        .then(reponse => console.log(response))
+    },
+    getFavorite(id) {
+      this.favorite ? (this.favorite = false) : (this.favorite = true)
+      let localUser = this.getLocalUser()
+      let favList = localUser.favorite_articles
```

Comment:
SO you can call yr computed here `let favList = this.localUser.favorite_articles`